Terminal.Gui/App/Application.Screen.cs

```diff
@@ -18,19 +19,27 @@ public static Rectangle Screen
     {
         get
         {
-            if (_screen == null)
+            lock (_lock)
             {
-                _screen = Driver?.Screen ?? new (new (0, 0), new (2048, 2048));
+                if (_screen == null)
+                {
+                    _screen = Driver?.Screen ?? new (new (0, 0), new (2048, 2048));
+                }
+
+                return _screen.Value;
             }
-            return _screen.Value;
         }
         set
         {
             if (value is {} && (value.X != 0 || value.Y != 0))
             {
                 throw new NotImplementedException ($"Screen locations other than 0, 0 are not yet supported");
             }
-            _screen = value;
```

Comment:
Another thing we can do here is check IsInitialized. Throw an exception if false. 

This will find the parallel test that is setting it.